### DISCUSSION TITLE
todo

### DISCUSSION THREAD

**Discussion by pkivolowitz** [pkivolowitz/asm_book#14] [Ideas]
1. Integrating with testing frameworks.
2. Integrating with Python.

**Comment by pkivolowitz** [comment]
https://realpython.com/python-bindings-overview/

**Comment by pkivolowitz** [comment]
3. Automatic PDF and HTML???

**Comment by multitudes** [comment]
in section_1/hello_world/README.md I could not make the hello word program work on my M1. I tried with the macros header but it did not help. On linux the small V5 program compiles with just cc. on mac I probably need to add some more complicated linking instructions because the puts and the c library are not found. Maybe could you add a code that would compile on mac too without jumping immediately into the macros? This would be useful. And then the compilation instructions would be nice. 
Since clang did not work I tried with `as -o hello_world.o hello_world` then with `ld` and a path to my mac sdk but still no success so far. 
[EDIT] Had to write `_puts` to make it working. like `_main`

**Reply by pkivolowitz** [reply]
Yes, that's right. Put underscores before puts and main. Apple.

I supposed I can provide a second version but the macros I wrote eliminate all this. Give me a few minutes.

Thank you for your input.

**Reply by pkivolowitz** [reply]
Your suggestion has been implemented with attribution made!

**Reply by multitudes** [reply]
Thank you for the quick reply, and the citation (it is not needed though! :) )  
I really like your methodical approach to make code for Linux and mac because I use both (I have intel Linux at the moment so I have to practice on the mac...) ! 
It is beautiful to see the code difference at a glance and then go into using the macros but having that background knowledge of how to code for mac when needed. I might send you more suggestions in the future :)
